### PR TITLE
Document --help better.

### PR DESCRIPTION
**Pull request by schveiguy**
Removed redundant documentation for `GetoptResult`

### CODE REVIEW

File: std/getopt.d [hunk 1]
```diff
@@ -413,7 +413,7 @@ getopt(args,
 An unrecognized option such as "--baz" will be found untouched in
 $(D args) after $(D getopt) returns.
 
-$(D Help Information Generation)
+$(B Help Information Generation)
```
Comment (andralex) [inline std/getopt.d]: Could you please make this and the other ones with the same role an H3 or H4?

Comment (schveiguy) [inline std/getopt.d]: Can do.

Comment (schveiguy) [inline std/getopt.d]: I looked throughout phobos, couldn't find an example that uses H3 or H4 (just grepped for H3 and H4). I found in std.process, it simply does things like

```D
/**
   Some docs
   Command_line:
   information about command line requirements
*/
```
Which I think would work out fine. Is there something specific you had in mind?

Comment (schveiguy) [inline std/getopt.d]: I'm going with that. I have absolutely zero idea how to build the phobos ddocs, so I can't give you a preview. There seems to be many cross dependencies between phobos and dlang.org, and I don't know how to set that up. I really REALLY think there should be a way to say "make html" and it works if you have some dependencies cloned from dlang.org there.

File: std/getopt.d [hunk 2]
```diff
@@ -298,8 +298,7 @@ void main(string[] args)
     ))
 )
 
-$(B Options with multiple names)
-
+Options_with_multiple_names:
```
Comment (burner) [inline std/getopt.d]: why did you remove all the bold face

File: std/getopt.d [hunk 3]
```diff
@@ -488,9 +483,7 @@ enum config {
 
 /** The result of the $(D getopt) function.
 
-The $(D GetoptResult) contains two members. The first member is a boolean with
-the name $(D helpWanted). The second member is an array of $(D Option). The
-array is accessable by the name $(D options).
+$(D helpWanted) is set if the option `--help` or `-h` was passed to the option parser.
```
Comment (burner) [inline std/getopt.d]: I'm not sure if less informatik is good here, as people may want to implement their own GetoptResult writer.

Comment (schveiguy) [inline std/getopt.d]: The struct members are fully documented. This literally adds zero extra knowledge, and reads like someone trying to fulfill a requirement to "document everything".

See: http://dlang.org/phobos/std_getopt.html#.GetoptResult

File: std/getopt.d (hunk 2 above)
Comment (schveiguy) [inline std/getopt.d]: Per Andrei's request. The usage of the colon section will make it very similar. Please see https://github.com/D-Programming-Language/phobos/pull/3347#discussion_r31442154

**Comment by burner** [comment]
LGTM

**Comment by burner** [comment]
Auto-merge toggled on
